config/cmake/modules/MfemCmakeUtilities.cmake

```diff
@@ -861,7 +861,7 @@ function(mfem_export_mk_files)
     endif()
   endforeach()
   # TODO: Add support for MFEM_USE_CUDA=YES
-  set(MFEM_CXX ${CMAKE_CXX_COMPILER})
+  set(MFEM_CXX ${CMAKE_CXX_COMPILER} CACHE STRING "" FORCE)
```

Comment:
Pushing variables to the cache in this function is not a good idea -- the values set in this function are specifically temporary, see the comment above, on line 843.

If any of these values are specifically needed for JIT, we can explicitly push them to the cache from here with names that add `JIT_` prefix or `_JIT` suffix.

Reply:
Actually, I take this back. None of the variables pushed to the cache in this function (`MFEM_CXX`, `MFEM_TPLFLAGS`, `MFEM_INCFLAGS`, `MFEM_LIBS`, `MFEM_EXT_LIBS`, `MFEM_BUILD_FLAGS`) have values before this function, so nothing is being overwritten.

However, it looks like some of them are not actually used in `MjitCmakeUtilities.cmake`: `MFEM_INCFLAGS`, `MFEM_LIBS`, so they probably do not need to be pushed to the cache.

Reply:
Right, I removed `MFEM_INCFLAGS` & `MFEM_LIBS`.